genomepy/annotation.py

```diff
@@ -0,0 +1,395 @@
+        """
+        Returns the filepath to a single (gzipped) file in the genome_dir with matching ext.
+        """
+        ext_files = glob_ext_files(self.genome_dir, ext)
+        fname = [f for f in ext_files if f"{self.name}.{ext}" in f]
+        if len(fname) == 1:
+            filepath = os.path.join(self.genome_dir, fname[0])
+            return filepath
+
+        # error handling
+        if len(fname) > 1:
+            raise IndexError(
+                f"Too many ({len(fname)}) files match '{self.name}.{ext}(.gz)' "
+                f"in directory {self.genome_dir}"
+            )
+        if check_exists:
+            raise FileNotFoundError(
+                f"could not find '{self.name}.{ext}(.gz)' in directory {self.genome_dir}"
+            )
+        return None
+
+    @property
+    def genome_contigs(self):
+        return self.__genome_contigs
+
+    @genome_contigs.setter
+    def genome_contigs(self, genome_contig_list):
+        self.__genome_contigs = genome_contig_list
+
+    @genome_contigs.getter
+    def genome_contigs(self):
+        if not self.__genome_contigs:
+            return get_column(self.sizes_file)
+
+    @property
+    def annotation_contigs(self):
+        return self.__annotation_contigs
+
+    @annotation_contigs.setter
+    def annotation_contigs(self, annotation_contig_list):
+        self.__annotation_contigs = annotation_contig_list
+
+    @annotation_contigs.getter
+    def annotation_contigs(self):
+        if not self.__annotation_contigs:
+            return list(set(get_column(self.annotation_bed_file)))
+
+    def filter_regex(
+        self,
+        annotation_file: str,
+        regex: Optional[str] = ".*",
+        invert_match: Optional[bool] = False,
+    ):
+        """
+        Filter annotation file (works with both gtf and bed) by contig name.
+
+        annotation_file: path to annotation file.
+        regex: regex string to keep.
+        invert_match: keep contigs NOT matching the regex string.
+
+        return a list of discarded contigs.
+        """
+        missing_contigs = []
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+        filtered_annotation_file = os.path.join(
+            tmp_dir, os.path.basename(annotation_file)
+        )
+        with _open(annotation_file, "r") as old, _open(
+            filtered_annotation_file, "w"
+        ) as new:
+            for line in old:
+                if bool(re.match(regex, line)) is not invert_match:
+                    new.write(line)
+                else:
+                    missing_contigs.append(line.split("\t")[0])
+
+        shutil.move(filtered_annotation_file, annotation_file)
+        rm_rf(tmp_dir)
+        return list(set(missing_contigs))
+
+    def filter_genome_contigs(self):
+        """
+        filter the gtf for contigs in the genome.
+        return a list of discarded contigs.
+        """
+        missing_contigs = []
+        genome_contigs = self.genome_contigs
```

Comment:
I think this is kinda "dangerous". `genome_contigs` is a list, which means it is mutable, which in turn means that the = does not mean it takes a copy. Both self.genome_contigs and genome_contigs are exactly the same variable, just with a different name. If you now start changing genome_contigs, it means you change self.genome_contigs.